### PR TITLE
Merging In things that CivEx uses and simple dupe fix

### PR DESCRIPTION
**Pull request by Ryan00793**
Dupe fix prevents anything that inherits InventoryHolder and prevents it from going through a portal. It also ignores players thanks to the event not throwing if a player triggers.

Should probably be squashed

This grew while talking about it in discord and is no longer just a simple dupe fix but alas it should fix things... probably... maybe... hopefully.

### CODE REVIEW

**Comment by ProgrammerDan** [comment]
These are pretty exciting

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFixes.java [hunk 1]
```diff
@@ -181,21 +194,22 @@ public void onRailPlace(BlockPlaceEvent event)
 			Block block = event.getBlock();
 			Material type = block.getType();
 
-			if(type == Material.RAILS ||
-				type == Material.ACTIVATOR_RAIL ||
-				type == Material.DETECTOR_RAIL ||
-				type == Material.POWERED_RAIL)
+			for (Material mat : stopDupes_)
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFixes.java]: why not just `if (stopDupes_.contains(mat)) {`?

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFixes.java [hunk 2]
```diff
@@ -162,12 +174,13 @@ public void onPistonPushRail(BlockPistonExtendEvent event)
 			for (Block block : event.getBlocks())
 			{
 				Material type = block.getType();
-				if(type == Material.RAILS ||
-					type == Material.ACTIVATOR_RAIL ||
-					type == Material.DETECTOR_RAIL ||
-					type == Material.POWERED_RAIL)
+				for (Material mat : stopDupes_)
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFixes.java]: same; `if (stopDupes_.contains(mat)) {`

assuming you change stopDupes from a flat array to a structured Set, ofc.

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/configs/GameFeaturesConfig.java [hunk 3]
```diff
@@ -46,6 +50,17 @@ protected void wireup(ConfigurationSection config) {
 
 		this.enderChestPlacement = config.getBoolean("enderChestPlacement", false);
 		if (!this.enderChestPlacement) plugin().log("Placeing EnderChests is disabled");
+
+		this.weepingAngel = config.getBoolean("weepingAngel.enabled", false);
+		if(this.weepingAngel)
+		{
+			this.weepingAngelEnv = config.getInt("weepingAngel.enviorment", 1);
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/configs/GameFeaturesConfig.java]: `enviorment` -> `environment` ?

File: src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFeatures.java [hunk 4]
```diff
@@ -179,4 +191,69 @@ public void disableEnderChestPlacement(BlockPlaceEvent event)
...
+
+		Date exp = DateUtils.addMinutes(new Date(), minutes);
+		Bukkit.getServer().getBanList(BanList.Type.NAME).addBan(p.getName(), "You've been banned for " + minutes +
+				" minutes due to your death.", exp, "weepingAngel");
+
+		Bukkit.getServer().getScheduler().scheduleSyncDelayedTask(SimpleAdminHacks.instance(), new Runnable()
+		{
+			public void run()
+			{
+				p.kickPlayer("You've been banned for " + minutes + " minutes due to your death.");
```
Comment (ProgrammerDan) [inline src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFeatures.java]: What handles unban?

File: src/main/resources/config.yml [hunk 5]
```diff
@@ -24,12 +24,19 @@ hacks:
     enderChestPlacement: false
     elytra: off
     chorus_fruit: off
+    # WeepingAngel bans people when they die, the numbers are in minutes.
+    weepingAngel:
+      enabled: false
+      enviorment: 1
```
Comment (ProgrammerDan) [inline src/main/resources/config.yml]: `enviorment` -> `environment`

**Comment by Ryan00793** [comment]
Pushed fixes

**Comment by ProgrammerDan** [comment]
This stuff looks great.

All tested?

**Pull request merged**